test/config-next/ca-b.json

```diff
@@ -83,7 +83,7 @@
         "allowMustStaple": true,
         "allowCTPoison": true,
         "allowSCTList": true,
-        "allowCommonName": true,
+        "allowCommonName": false,
```

Comment:
`allowCommonName: false` [1] and `SetCommonName: true` [2] seem like they're going to conflict and I don't think there's a test in `//issuance/issuance_test.go` for it.

[1] https://github.com/letsencrypt/boulder/blob/e052e7445bebbe4c9062235bd46931cf936576b7/issuance/issuance.go#L282-L284
[2] https://github.com/letsencrypt/boulder/pull/6706/files#diff-9033031dd64a1f3c17bc6f97c5e0c3a0375eafa6e81bbbbbaccedc6446d94c8dR446-R448

Reply:
Yes, it is correct that having `"SetCommonName": true` and `"allowCommonName": false` at the same time would lead to errors. But we never expect to enter that state. The rollout can follow either of two paths:
1) First set `"SetCommonName": false`, then in a later deployment set `"allowCommonName: false`; or
2) Set both to `false` at the same time, because they're in the same config file (what this PR does).

I've added a unittest for `"allowCommonName": false`, though.